iridium-client-nextjs/src/lib/services/exchange-URL-generator.service.ts

```diff
@@ -0,0 +1,30 @@
+import { CookieService } from './cookie.service';
+import { OauthConstants } from './oauth-constants';
+import { exchange } from './accessToken.service';
+
+export class ExchangeURLGeneratorService {
+    generate() {
+        const queryString = window.location.search;
+        const urlParams = new URLSearchParams(queryString);
+        const output = this.authorize();
+        return process.env.NEXT_PUBLIC_IRIDIUM_DOMAIN + 'oauth/token?grant_type=authorization_code&code=' + urlParams.get('code') + '&redirect_uri=' + output?.redirectUri + '&client_id=' + output?.clientId + '&code_verifier=' + output?.codeVerifier + '&state=' + output?.state;
```

Comment:
String concat that is this long is nasty business. Personally I'd use a template literal so I don't have to worry about missing a plus and its a "little" more readable

``${
  process.env.NEXT_PUBLIC_IRIDIUM_DOMAIN
}oauth/token?grant_type=authorization_code&code=${urlParams.get(
  'code'
)}&redirect_uri=${output?.redirectUri}&client_id=${
  output?.clientId
}&code_verifier=${output?.codeVerifier}&state=${output?.state}`;`

Reply:
`

`${process.env.NEXT_PUBLIC_IRIDIUM_DOMAIN }oauth/token?grant_type=authorization_code&code=${urlParams.get( 'code' )}&redirect_uri=${output?.redirectUri}&client_id=${ output?.clientId }&code_verifier=${output?.codeVerifier}&state=${output?.state}`;

`